database/DevHome.Database/Services/RepositoryManagementDataAccessService.cs

```diff
@@ -0,0 +1,228 @@
+            {
+                _log.Information($"No file exists at {configurationFileLocationAndName}.  This repository will not have a configuration file.");
+            }
+            else
+            {
+                newRepo.ConfigurationFileLocation = configurationFileLocationAndName;
+            }
+        }
+
+        try
+        {
+            using var dbContext = _databaseContextFactory.GetNewContext();
+            dbContext.Add(newRepo);
+            dbContext.SaveChanges();
+        }
+        catch (Exception ex)
+        {
+            _log.Error(ex, $"Exception when saving in {nameof(MakeRepository)}");
+            TelemetryFactory.Get<ITelemetry>().Log(
+                "DevHome_Database_Event",
+                LogLevel.Critical,
+                new DevHomeDatabaseEvent(nameof(MakeRepository), ex));
+            return new Repository();
+        }
+
+        return newRepo;
+    }
+
+    public List<Repository> GetRepositories()
+    {
+        _log.Information("Getting repositories");
+        List<Repository> repositories = [];
+
+        try
+        {
+            using var dbContext = _databaseContextFactory.GetNewContext();
+            repositories = [.. dbContext.Repositories];
+        }
+        catch (Exception ex)
+        {
+            _log.Error(ex, ex.ToString());
+            TelemetryFactory.Get<ITelemetry>().Log(
+                "DevHome_Database_Event",
+                LogLevel.Critical,
+                new DevHomeDatabaseEvent(nameof(GetRepositories), ex));
+        }
+
+        return repositories;
+    }
+
+    public Repository? GetRepository(string repositoryName, string cloneLocation)
+    {
+        _log.Information("Getting a repository");
+        try
+        {
+            using var dbContext = _databaseContextFactory.GetNewContext();
+#pragma warning disable CA1309 // Use ordinal string comparison
+            return dbContext.Repositories.FirstOrDefault(x => x.RepositoryName!.Equals(repositoryName)
+            && string.Equals(x.RepositoryClonePath, Path.GetFullPath(cloneLocation)));
+#pragma warning restore CA1309 // Use ordinal string comparison
+        }
+        catch (Exception ex)
+        {
+            _log.Error(ex, ex.ToString());
+            TelemetryFactory.Get<ITelemetry>().Log(
+                "DevHome_Database_Event",
+                LogLevel.Critical,
+                new DevHomeDatabaseEvent(nameof(GetRepository), ex));
+        }
+
+        return null;
+    }
+
+    public bool UpdateCloneLocation(Repository repository, string newLocation)
+    {
+        try
+        {
+            using var dbContext = _databaseContextFactory.GetNewContext();
+            var maybeRepository = dbContext.Repositories.Find(repository.RepositoryId);
+            if (maybeRepository == null)
+            {
+                _log.Warning($"{nameof(UpdateCloneLocation)} was called with a RepositoryId of {repository.RepositoryId} and it does not exist in the database.");
```

Comment:
This is unexpected behavior.  I decided on `warning` because calling `UpdateCloneLocation` with a repository that isn't in the database is a usage issue, not an error.